conda-env-mod

```diff
@@ -129,8 +128,14 @@ function create_env()
 	#
 	# Now the actual conda create
 	#
-	echo conda create -q --copy "${CONDA_ARGS[@]}" python="${PY}" "${CONDA_XTRA[@]}"
-	conda create -q --copy "${CONDA_ARGS[@]}" python="${PY}" "${CONDA_XTRA[@]}"
+	# Check yaml file
+	if [[ $Opt_YAML -ne 0 ]]; then
+		echo conda env create -q "${CONDA_ARGS[@]}" python="${PY}" "${CONDA_XTRA[@]}"
+		conda env create -q "${CONDA_ARGS[@]}" python="${PY}" "${CONDA_XTRA[@]}"
```

Comment:
Do we need to specify the Python version when using `--file`? Won't the environment file have python in the list? What happens if it doesn't?

Reply:
I LITERALLY had the same lines highlighted and started to write the same thing last night :)

I too was contemplating dropping the `python="${PY}"` for the case of `--file`... no firm opinion except that _sometimes_ it can help to make sure that environment name matches the actual content (when python is not listed in the YAML).  

But at the same time the big concern is that it may override what's in the YAML (so suddenly the resulting environment would be clandestinely different from what the researcher expected, which is bad for reproducibility).

> Won't the environment file have python in the list? What happens if it doesn't?

An equally interesting question is also what happens if it _does_?  Which version wins - the $PY one or the one inside the YAML?  Are command-line packages even considered when `--file` is in effect?

Sounds like an experiment to be had :)
|                     | Python version in YAML | No Python version in YAML |
|---------------------|------------------------|---------------------------|
| With python=${PY}   |                        |                           |
| Without python=${PY} |                        |                           |